main.py

```diff
@@ -554,4 +686,35 @@ def scsyntheticunithydrograph(request_body: SCSyntheticUnitHydrograph, response:
         }
 
     except Exception as e:
-        raise HTTPException(status_code = 500, detail =  str(e))
\ No newline at end of file
+        raise HTTPException(status_code = 500, detail =  str(e))
+
+@app.post("/calculatemissingparameters/")
+def calculatemissingparameters(request_body: CalculateMissingParameters, response: Response):
+
+    try: 
+        rainfall_distribution_curve_letter, Tc, PRF, CN, S, Ia = calculateMissingParameters(
+            request_body.lat,
+            request_body.lon,
+            request_body.AEP,
+            request_body.curveNumberMethod,
+            request_body.TcMethod,        
+            request_body.length,
+            request_body.slope,      
+            request_body.dataSheetFlow,
+            request_body.dataExcessSheetFlow,
+            request_body.dataShallowConcentratedFlow,
+            request_body.dataChannelizedFlowOpenChannel,
+            request_body.dataChannelizedFlowStormSewer,
+            request_body.dataChannelizedFlowStormSewerOrOpenChannelUserInputVelocity,
+        )
+        return {
+            "rainfall_distribution_curve_letter": rainfall_distribution_curve_letter,
+            "time_of_concentration": Tc,
```

Comment:
I got the following response body:

```
{
  "rainfall_distribution_curve_letter": "B",
  "time_of_concentration": [
    64.46157606015156,
    null
  ],
  "peak_rate_factor": 240,
  "curve_number": 66.9098326632795,
  "S": 4.945486488248321,
  "Ia": 0.9890972976496641
}
```

Do you know why Tc is returning an array with `null`? I think it should just be a single number. Apologies if this is due to an error I caused.

Reply:
Its the warning message being returned: https://github.com/USGS-WiM/SC-RunoffModelingServices/blob/ad5203ba08a39397e0daa0df7d857c7c2486c3c8/Tc_Calculator.py#L255. I wasn't sure if I should grab that or not.

Reply:
I could create a separate return variable for the warning message, that might be easier when it comes to working with it in the client?

Reply:
Oh sorry, I was looking at the wrong function return 🤦‍♀️. Let's keep it as-is.

Reply:
Added warningMessage.